packages/boot/tools/ibc/mocks.js

```diff
@@ -1,3 +1,69 @@
+/** @import { Base64Bytes, ChannelOrdering, DestinationPortID, SourcePortID } from '@agoric/orchestration'; */
+/** @import { IBCConnectionID, IBCPacket, IBCPortID } from '@agoric/vats'; */
+
+/**
+ * based on BridgeId.DIBC events
+ * @typedef {{
+ *  hops: IBCConnectionID[];
+ *  method: 'startChannelOpenInit';
+ *  order: ChannelOrdering;
+ *  packet: {
+ *    destination_port: IBCPortID | DestinationPortID;
+ *    source_port: IBCPortID | SourcePortID;
+ *  };
+ *  type: 'IBC_METHOD';
+ *  version: string;
+ * }} ChannelOpenInitPacket
```

Comment:
this typedef is not only a concern of mocks, right? Please put it somewhere more canonical

Reply:
Noted. This led to a bigger PR, but seems worthwhile. Some changes are in `7406cef` and `e49a837`. Expect one more commit to remove the `IBCDowncall<E>` type casting in favor of a parameter to `.downcall()`